# Acuerdos de paz en Colombia entre 1990 y 2017 con Fuerzas Armadas Revolucionarias de Colombia (FARC-EP)

En este dodumento se estudiarán todos los acuerdos de paz en Colombia desde 1990 hasta 2017 on las Fuerzas Armadas Revolucionarias de Colombia (FARC-EP) viendo la evolución de estos.
Se ha desarrollado este estudio con la finalidad de aclarar y dar un poco de luz a los últimos año de una de las guerras civiles más largas y sangrientas del último siglo. 
La fuente utilizada es **peace agreements database** de la universidad de Edimburgo.

Este estudio ha sido desarrollado en Salamamanca (España), el 19 de Abril de 2020 por Emilio José Lucas Marcos.

In [6]:
import pandas as pd

In [7]:
df = pd.read_csv('pax_data_129_agreements_18-04-20.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 129 entries, 0 to 128
Columns: 265 entries, Con to ImSrc
dtypes: float64(3), int64(247), object(15)
memory usage: 267.2+ KB


In [8]:
df.dtypes

Con       object
Contp     object
PP         int64
PPName    object
Reg       object
           ...  
ImOth      int64
ImRef      int64
ImPK       int64
ImE        int64
ImSrc      int64
Length: 265, dtype: object

El conjunto de datos está compuesto por 265 variables (248 numéricas, 16 cadenas de texto y 1 fecha) y 129 filas.
Del las variables númericas la gran mayoría son variables binárias de 1 y 0 que serán convertidas a booleanas.
Los númericos que no se puedan considerar como tal, por ejemplo, el número de conflicto, pasarán a ser de tipo cadena de texto.
También se eliminarán las variables que sean igual en todas las filas ya que no aportarán valor al estudio y las variables que aporten la misma información dos veces.
Además, se eliminarán variables que no sirvan para el estudio como la longitud de los documentos.
Se seleccionarán solo los tipos de conflicto interno.
También se eliminarán variables que no se puedan utilizar en la visualización como por ejemplo las que están sacadas directamente de los acuerdos y es un texto de un párrafo.

In [9]:
print(len(df.columns))
nunique = df.apply(pd.Series.nunique)
cols_to_drop = nunique[nunique == 1].index
df.drop(cols_to_drop, axis=1, inplace=True)
print(len(df.columns))
dataTypeDict = dict(df.dtypes)
dfCualitativas = pd.DataFrame()
dfCuantitativas = pd.DataFrame()
for tipo in dataTypeDict:
    if dataTypeDict[tipo] == 'object':
        dfCualitativas[tipo] = df[tipo]
    else:
        dfCuantitativas[tipo] = df[tipo]
df['Dat']= pd.to_datetime(df['Dat'])
df.head()

265
219


Con       Contp   PP                                          PPName  \
0  Colombia  Government  144  Colombia VI - Government-ELN post-2015 process   
1  Colombia  Government  144  Colombia VI - Government-ELN post-2015 process   
2  Colombia  Government  144  Colombia VI - Government-ELN post-2015 process   
3  Colombia  Government  144  Colombia VI - Government-ELN post-2015 process   
4  Colombia  Government  144  Colombia VI - Government-ELN post-2015 process   

   AgtId                                                Agt        Dat  \
0   2045  Joint Announcement by the National Government ... 2017-10-24   
1   2043  Comunicado Conjunto 5: Entre El Gobierno Nacio... 2017-09-24   
2   1890           Comunicado Conjunto 4: Acuerdo de Quito  2017-09-04   
3   1937                              Comunicado Conjunto 3 2017-06-06   
4   1936                              Comunicado Conjunto 2 2017-04-06   

                     Status  Lgt  N_characters  ... TjMis TjRep TjRSym TjRMa  \
0  Multiparty signed/agreed    2          4138  ...     0     0      0     0   
1  Multiparty signed/agreed    2          4464  ...     0     0      0     0   
2  Multiparty signed/agreed    1          1453  ...     0     0      0     0   
3  Multiparty signed/agreed    1          1695  ...     0     0      0     0   
4  Multiparty signed/agreed    1          1561  ...     0     0      0     0   

  TjNR ImUN  ImOth  ImRef  ImPK  ImE  
0    0    0      0      0     0    1  
1    0    0      0      0     0    1  
2    0    0      0      0     0    1  
3    0    0      0      0     0    1  
4    0    0      0      0     0    1  

[5 rows x 219 columns]